Resources/public/js/views/cof-contentcreationview.js

```diff
@@ -0,0 +1,208 @@
+        },
+
+        render: function () {
+            this.get('container').setHTML(this.template({
+                finishBtnText: this.get('finishBtnText')
+            }));
+
+            return this;
+        },
+
+        /**
+         * Enables the next button
+         *
+         * @protected
+         * @method _enableNextButton
+         * @param event {Object} event facade
+         */
+        _enableNextButton: function (event) {
+            var container = this.get('container');
+
+            container.one(SELECTOR_CHANGE_CONTENT_TYPE).setHTML(event.text);
+            container.one(SELECTOR_NEXT_BUTTON).removeClass(CLASS_BUTTON_DISABLED);
+        },
+
+        /**
+         * Gets the Content Types
+         *
+         * @protected
+         * @method _getContentTypes
+         * @param event {Object} event facade
+         */
+        _getContentTypes: function (event) {
+            var container = this.get('container');
+
+            if (!event.newVal) {
+                this._resetFormState();
+
+                return;
+            }
+
+            this.get('contentTypeSelectorView').hideTooltip();
+
+            container.one(SELECTOR_CONTENT_TYPE_SELECTOR).addClass(CLASS_SELECTOR_LOADING);
+            container.one(SELECTOR_NEXT_BUTTON).addClass(CLASS_BUTTON_DISABLED);
+
+            /**
+             * Fetches the content types data.
+             * Listened in the cof.Plugin.createContentSelectContentType
+             */
+            this.fire('fetchContentTypes');
+        },
+
+        /**
+         * Resets the content creation form state
+         *
+         * @protected
+         * @method _resetFormState
+         */
+        _resetFormState: function () {
+            this.set('activePage', 0);
+
+            this.get('container').removeClass(CLASS_ON_PAGE_TWO);
+        },
+
+        /**
+         * Render the Content Type Selector View.
+         *
+         * @protected
+         * @method _renderContentTypeSelector
+         * @param event {Object} event facade
+         */
+        _renderContentTypeSelector: function (event) {
+            var contentTypeSelector = this.get('container').one(SELECTOR_CONTENT_TYPE_SELECTOR),
+                contentTypeSelectorView = this.get('contentTypeSelectorView');
+
+            contentTypeSelectorView.set('contentTypeGroups', event.newVal);
+
+            contentTypeSelector.append(contentTypeSelectorView.render().get('container'));
+            contentTypeSelector.removeClass(CLASS_SELECTOR_LOADING);
+
+            contentTypeSelectorView.showTooltip();
+        },
+
+        /**
+         * Changes the page of the Content Creation Widget.
+         *
+         * @protected
+         * @method _changeFormPage
+         * @param event {Object} event facade
+         */
+        _changeFormPage: function (event) {
+            var activePage = this.get('activePage'),
+                transformMethodName = activePage ? 'removeClass' : 'addClass',
```

Comment:
The variable name is misleading. I would rather expect something else than just adding/removing classes.